Makefile

```diff
@@ -315,9 +315,40 @@ ifeq ('$(browser)','chrome-mv3')
   BUILD_TARGETS += $(BUILD_DIR)/data/bundled/smarter-encryption-rules.json
 endif
 
+## Generate Tracker Blocking declarativeNetRequest rulesets for MV3 builds.
+MV3_BLOCKLIST_TARGETS = $(INTERMEDIATES_DIR)/current-mv3-tds.json $(INTERMEDIATES_DIR)/fallback-mv3-tds.json
+MV3_CURRENT_RULESET_TARGETS = $(BUILD_DIR)/data/bundled/current-tds-rules.json $(BUILD_DIR)/data/bundled/current-ctl-allowing-rules.json
+MV3_FALLBACK_RULESET_TARGETS = $(BUILD_DIR)/data/bundled/fallback-tds-rules.json $(BUILD_DIR)/data/bundled/fallback-ctl-allowing-rules.json
+
+.SECONDARY:
+$(MV3_BLOCKLIST_TARGETS) &: shared/data/etags.json
+	node scripts/fetchMV3Blocklists.mjs
+	touch $(MV3_BLOCKLIST_TARGETS)
+
+$(BUILD_DIR)/data/bundled/current-mv3-tds.json: $(INTERMEDIATES_DIR)/current-mv3-tds.json
+	cp $< $@
+
+$(BUILD_DIR)/data/bundled/fallback-mv3-tds.json: $(INTERMEDIATES_DIR)/fallback-mv3-tds.json
+	cp $< $@
+
+$(MV3_CURRENT_RULESET_TARGETS) &: $(INTERMEDIATES_DIR)/current-mv3-tds.json $(INTERMEDIATES_DIR)/surrogates.json
+	npx ddg2dnr tds $^ $(MV3_CURRENT_RULESET_TARGETS)
+
+$(MV3_FALLBACK_RULESET_TARGETS) &: $(INTERMEDIATES_DIR)/fallback-mv3-tds.json $(INTERMEDIATES_DIR)/surrogates.json
+	npx ddg2dnr tds $^ $(MV3_FALLBACK_RULESET_TARGETS)
```

Comment:
These commands launch `chromium` for me during the build, which was unexpected. Is that meant to happen? If so, can we avoid it?

Reply:
But the `npx ddg2dnr smarter-encryption $< $@` line doesn't for you?

Reply:
Oh right, I remember. We're using Chrome via Puppeteer to access the [isRegexSupported](https://developer.chrome.com/docs/extensions/reference/declarativeNetRequest/#method-isRegexSupported) API. That's non-trivial to duplicate ourselves.